src/query.ts

```diff
@@ -674,6 +674,7 @@ export class Query<
       this._finished ||
       data.done ||
       !autoFetch ||
+      this.records.length === maxFetch ||
```

Comment:
`maxFetch` is always set (if user doesn't set it, it defaults to `10_000`). 

set finished=true if we already got enough records in the first request.